source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
It is necessary either to prescribe pressures or ignore the residual in the kinematically prescribed regions. Because there is a discontinuity in the prescribed velocity along the subduction interface, very large spurious pressures develop here and the solver will fail to converge. After prescribing pressure within the kinematically prescribed regions, we have no convergence issues. This insight was based on my past experiences with my finite volume code.